src/keystore/InMemoryKeystore.ts

```diff
@@ -272,9 +294,46 @@ export default class InMemoryKeystore implements Keystore {
           'missing recipient'
         )
       }
-      const invitation = InvitationV1.createRandom(req.context)
       const created = nsToDate(req.createdNs)
       const recipient = toSignedPublicKeyBundle(req.recipient)
+
+      const secret = await this.v2Keys.sharedSecret(
+        recipient,
+        this.v2Keys.getCurrentPreKey().publicKey,
+        false
+      )
+
+      const sortedAddresses = [
+        this.accountAddress,
+        await recipient.walletSignatureAddress(),
+      ].sort()
+
+      const msgString =
+        (req.context?.conversationId || '') + sortedAddresses.join()
+
+      const msgBytes = new TextEncoder().encode(msgString)
+
+      const topic = sha256(
```

Comment:
Just checking, this does a redundant `sha256` of the post `hmacSha256Sign` output, which is already run through `sha256`. Did we spec that out or otherwise can we remove it? Not a huge deal either way TBH but don't want non-spec operations sneaking in if we want to reproduce this on other platforms.

HMAC construction being: `HASH(Key XOR opad, HASH(Key XOR ipad, text))`

More details: https://security.stackexchange.com/a/79579

Reply:
Fixed in https://github.com/xmtp/xmtp-js/pull/326/commits/19a4f65de3a0e0b9d0695560a2f5d721b3600a91!